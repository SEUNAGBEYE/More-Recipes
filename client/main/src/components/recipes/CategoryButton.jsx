import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';


/**
 * @description - Component for selecting category to display
 * @param {Object} props
 *
 * @returns {jsx} JSX
 */
const CategoryButton = (props) => (
  <div className="dropdown" style={{ float: 'left' }}
    data-category-button="category-button"
  >
    <button className="btn btn-default dropdown-toggle auth-button"
      type="button" id="about-us" data-toggle="dropdown"
      aria-haspopup="true" aria-expanded="false">Category
    </button>
    <div className="dropdown-menu" aria-labelledby="about-us">
      {
        props.categories.map(category =>
          (<Link className="dropdown-item"
            key={category.id} to={`/categories/${category.name}`}
            data-category-link={`category-link-${category.name}`}
          >{category.name}</Link>))
      }
    </div>
  </div>
);

const propTypes = {
  categories: PropTypes.array.isRequired,
};

CategoryButton.propTypes = propTypes;

/**
 * mapStateToProps
 * @export
 * @param {Object} state
 *
 * @returns {Object} object
 */
export const mapStateToProps = (state) => ({
  categories: state.recipes.recipeCategories
    .map(category => ({ name: category.name, id: category.id }))
});

export { CategoryButton as CategoriesButton };
export default connect(mapStateToProps)(CategoryButton);
