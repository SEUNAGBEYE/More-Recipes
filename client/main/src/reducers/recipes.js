import actionTypes from '../actions/actionTypes';

const {
  GET_RECIPE,
  GET_RECIPES,
  GET_USER_RECIPES,
  GET_USER_FAVOURITED_RECIPES_ID,
  GET_FAVOURITED_RECIPES,
  GET_POPULAR_RECIPES,
  GET_RECIPE_REVIEWS,
  RECIPE_CATEGORIES,
  SEARCH_RECIPES,
  ADD_RECIPE,
  DELETE_RECIPE,
  EDIT_RECIPE,
  TOGGLE_FAVOURITE_RECIPE,
  TOGGLE_THUMBS_UP_RECIPE,
  TOGGLE_THUMBS_DOWN_RECIPE,
  REVIEW_RECIPE
} = actionTypes;

export const initialState = {
  allRecipes: [],
  singleRecipe: {
    id: '',
    name: '',
    image: '',
    categoryId: '',
    userId: '',
    upvotes: [],
    downvotes: [],
    views: [],
    description: '',
    reviews: []
  },
  pagination: '',
  userFavouritedRecipeId: [],
  favouriteRecipes: [],
  popularRecipes: [],
  recipeCategories: [],
  loaded: false
};

export default (state = initialState, action = {}) => {
  const singleRecipe = { ...state.singleRecipe };
  let allRecipes = [...state.allRecipes];
  const { recipe: recipeOnAction } = action;

  switch (action.type) {
  case ADD_RECIPE:
    return {
      ...state,
      ...{ allRecipes: [action.recipe, ...state.allRecipes] }
    };

  case GET_RECIPE:
    return {
      ...state,
      ...{
        singleRecipe: action.recipe,
        pagination: action.pagination,
        loaded: true
      }
    };

  case GET_USER_RECIPES:
    return {
      ...state,
      ...{
        allRecipes: action.recipes,
        pagination: action.pagination
      }
    };

  case GET_RECIPES:
    return {
      ...state,
      ...{
        allRecipes: action.recipes,
        pagination: action.pagination,
        loaded: true
      }
    };

  case RECIPE_CATEGORIES:
    return {
      ...state,
      ...{
        recipeCategories: action.recipeCategories,
        loaded: true
      }
    };
  case GET_POPULAR_RECIPES:
    return {
      ...state,
      ...{
        popularRecipes: [...action.popularRecipe],
        loaded: true
      }
    };

  case SEARCH_RECIPES:
    return {
      ...state,
      ...{
        allRecipes: [...action.recipes],
        pagination: action.pagination
      }
    };

  case DELETE_RECIPE:
    return {
      ...state,
      ...{
        allRecipes: state.allRecipes
          .filter(recipe => parseInt(recipe.id, 10) !== parseInt(action.id, 10))
      }
    };


  case EDIT_RECIPE:
    if (allRecipes.length >= 1) {
      allRecipes = allRecipes.map((recipe) => {
        if (parseInt(recipe.id, 10) === parseInt(action.recipe.id, 10)) {
          return action.recipe;
        }
        return recipe;
      });
    }

    return {
      ...state,
      ...{
        allRecipes,
        singleRecipe: { ...singleRecipe, ...recipeOnAction }
      }
    };

  case TOGGLE_FAVOURITE_RECIPE:
    return {
      ...state,
      ...{
        favouriteRecipes: !state.userFavouritedRecipeId
          .includes(action.favouritedRecipe.id) ?
          [...state.favouriteRecipes, action.favouritedRecipe] :
          state.favouriteRecipes
            .filter(recipe => recipe.id != action.favouritedRecipe.id),

        userFavouritedRecipeId: !state.userFavouritedRecipeId
          .includes(action.favouritedRecipe.id) ?
          state.userFavouritedRecipeId.concat(action.favouritedRecipe.id) :
          state.userFavouritedRecipeId
            .filter(id => id !== action.favouritedRecipe.id),
        singleRecipe: action.favouritedRecipe
      }
    };

  case TOGGLE_THUMBS_UP_RECIPE:
    return {
      ...state,
      ...{
        allRecipes: [...state.allRecipes.map((recipe) => {
          if (recipe.id === action.recipe.id) {
            return action.recipe;
          }
          return recipe;
        })],
        favouriteRecipes: [...state.favouriteRecipes.map((recipe) => {
          if (recipe.id === action.recipe.id) {
            return action.recipe;
          }
          return recipe;
        })],
        singleRecipe: action.recipe
      }
    };

  case TOGGLE_THUMBS_DOWN_RECIPE:
    return {
      ...state,
      ...{
        allRecipes: [...state.allRecipes.map((recipe) => {
          if (recipe.id === action.recipe.id) {
            return action.recipe;
          }
          return recipe;
        })],
        favouriteRecipes: [...state.favouriteRecipes.map((recipe) => {
          if (recipe.id === action.recipe.id) {
            return action.recipe;
          }
          return recipe;
        })],
        singleRecipe: action.recipe
      }
    };

  case GET_FAVOURITED_RECIPES:
    return {
      ...state,
      favouriteRecipes: [...action.favouriteRecipes],
      pagination: action.pagination
    };

  case GET_USER_FAVOURITED_RECIPES_ID:
    return {
      ...state,
      ...{ userFavouritedRecipeId: action.favouritedRecipesIds }
    };
  case REVIEW_RECIPE:
    singleRecipe.reviews = [action.reviewBody, ...singleRecipe.reviews];
    return {
      ...state,
      ...{ singleRecipe }
    };
  case GET_RECIPE_REVIEWS:
    singleRecipe.reviews = [...singleRecipe.reviews, ...action.reviews];
    return {
      ...state,
      ...{
        singleRecipe,
        pagination: action.pagination
      }
    };

  default: return {
    ...state,
    ...{ loaded: true }
  };
  }
};

export { actionTypes };
