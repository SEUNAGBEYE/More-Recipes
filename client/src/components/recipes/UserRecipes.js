import React, { Component } from 'react';
import { Link } from 'react-router-dom'
import Navbar from '../common/Navbar';
import Footer from '../common/Footer';
import Recipe from '../../actions/Recipes';
import RecipeCard from './RecipeCard';
import Pagination from './Pagination';

/**
 * @class UserRecipes
 */
export default class UserRecipes extends Component{

    state = {
      times: [...'Univelcityyy']
    }

  /**
   * @memberOf UserRecipes
   * return {object}
   */
  render(){
    return(
      <div>
        <Navbar />
        <main style={{marginTop: 40}}>

          <div className="container">
            <div className="dropdown" style={{float: 'left'}}>
              <button className="btn btn-default dropdown-toggle auth-button" type="button" id="about-us" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">Category
              </button>
              <div className="dropdown-menu" aria-labelledby="about-us">
                <a className="dropdown-item" href="recipes.html">Dessert</a>
                <a className="dropdown-item" href="my_recipes.html">Pasta</a>
                <a className="dropdown-item" href="favourite_recipe.html">Fries</a>
                <a className="dropdown-item" href="#">Chinese</a>
                <a className="dropdown-item" href="index.html">Africa</a>
              </div>
            </div>
          </div>
          
          <div className="container">
            <div style={{textAlign: 'center', marginTop: 100}}>
              <a href="" className="auth-button" data-toggle="modal" data-target="#editModal">Add Recipe</a>
              <h4 className='container__myrecipes'>My Recipes</h4><br /><br />
            </div>

            <div className='row'>
              {
                this.state.times.map((elem, index) => {
                return (<RecipeCard key={index}/>)
                })
              }
            </div>
          </div>
        </main>
        <Pagination />
        <Footer />
      </div>
    )
  }
}

