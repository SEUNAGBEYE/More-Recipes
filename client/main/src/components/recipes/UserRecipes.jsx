import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { addRecipe,
  getUserRecipes,
  recipeCategories }
  from '../../actions/Recipes';
import RecipeCard from './RecipeCard';
import Pagination from './Pagination';
import Exclamation from './Exclamation';
import RecipeModal from './RecipeModal';
import CategoryButton from './CategoryButton';


/**
 * @export
 * @class UserRecipes
 * @extends {Component}
 */
export class UserRecipes extends Component {
  /**
   * @description - Creates an instance of UserRecipes.
   *
   * @method constructor
   *
   * @param {Object} props
   *
   * @returns {void}
   *
   * @memberof UserRecipes
   */
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      description: '',
      image: '',
      ingredients: [],
      steps: [],
      errors: {},
      categoryId: '',
      downvotes: [],
      upvotes: [],
      userRecipes: []
    };
    this.paginateRecipe = this.paginateRecipe.bind(this);
  }


  /**
   * @description - Fetch Component Data
   *
   * @method componentDidMount
   *
   * @returns {void} void
   * @memberof UserRecipes
   */
  componentDidMount() {
    this.props.recipeCategories();
    this.paginateRecipe();
  }

  /**
   * @description - Paginate Component Data
   *
   * @method paginateRecipe
   *
   * @param {Number} [page=1]
   *
   * @returns {void} void
   * @memberof UserRecipes
   */
  paginateRecipe(page = 1) {
    this.props.getUserRecipes(page)
      .then(res => {
        this.setState({
          userRecipes: [...this.state.userRecipes, ...res.recipes]
        });
      });
  }


  /**
   * @description - Renders react component
   *
   * @method render
   *
   * @returns {Jsx} Jsx
   * @memberof UserRecipes
   */
  render() {
    return (
      <div id="userRecipesBody">
        <main style={{ marginTop: 40 }}>

          <div className="container">
            <CategoryButton />
          </div>

          <div className="container">
            <div style={{ textAlign: 'center', marginTop: 100 }}>
              <Link to="#" className="btn auth-button"
                data-toggle="modal"
                id="addRecipe"
                data-target="#addModal">Add Recipe</Link>
              <h4 className="container__myrecipes">My Recipes</h4><br /><br />
              <RecipeModal addRecipe={this.props.addRecipe}
                recipeCategories={this.props.categories}/>
            </div>

            <div className="row">
              {
                this.props.userRecipes.length > 0 ?
                  this.props.userRecipes.map((elem, index) => (
                    <RecipeCard key={elem.id}
                      recipe={elem} id={elem.id}
                      history={this.props.history}
                    />)) :
                  <Exclamation>
                    <p className="text-muted">Sorry you haven't added any
                     recipe yet, please add to get started</p>
                  </Exclamation>
              }
            </div>
          </div>
        </main>
        {this.props.pagination > 1 ?
          <Pagination recipesCount={this.props.pagination}
            recipesPagination={this.paginateRecipes}/> :
          ''
        }
      </div>
    );
  }
}

const propTypes = {
  pagination: PropTypes.number.isRequired,
  categories: PropTypes.array.isRequired,
  user: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
  userRecipes: PropTypes.array.isRequired,
  addRecipe: PropTypes.func.isRequired,
  getUserRecipes: PropTypes.func.isRequired,
  recipeCategories: PropTypes.func.isRequired,
};

UserRecipes.propTypes = propTypes;

/**
 * mapStateToProps
 * @param {Object} state
 *
 * @returns {Object} object
 */
export const mapStateToProps = (state) => ({
  recipes: state.recipes,
  user: state.auth.user,
  userRecipes: state.recipes.allRecipes,
  pagination: Number(state.recipes.pagination),
  categories: state.recipes.recipeCategories
});

export default connect(mapStateToProps, {
  addRecipe, getUserRecipes, recipeCategories
})(UserRecipes);

