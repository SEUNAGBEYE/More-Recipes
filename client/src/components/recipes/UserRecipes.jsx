import React, { Component } from 'react';
import { connect } from 'react-redux';
import { addRecipe,
  getUserRecipes,
  recipeCategories }
  from '../../actions/Recipes';
import RecipeCard from './RecipeCard';
import Pagination from './Pagination';
import Exclamation from './Exclamation';
import RecipeModal from './RecipeModal';
import CategoryButton from './CategoryButton';


/**
 * @class UserRecipes
 */
class UserRecipes extends Component {
  /**
   * Creates an instance of UserRecipes.
   * @param {any} props
   * @memberof UserRecipes
   */
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      description: '',
      image: '',
      ingredients: [],
      steps: [],
      errors: {},
      categoryId: '',
      downvotes: [],
      upvotes: [],
      userRecipes: []
    };
  }

  /**
  * @returns {void} void
  * @memberof UserRecipes
  */
  componentDidMount() {
    this.props.recipeCategories();
    this.props.getUserRecipes()
      .then(res => {
        this.setState({
          userRecipes: [...this.state.userRecipes, ...res.recipes]
        });
      });
  }


  /**
   * @memberOf UserRecipes
   * @returns {jsx} JSX
   */
  render() {
    return (
      <div>
        <main style={{ marginTop: 40 }}>

          <div className="container">
            <CategoryButton />
          </div>

          <div className="container">
            <div style={{ textAlign: 'center', marginTop: 100 }}>
              <a href="" className="auth-button"
                data-toggle="modal"
                data-target="#addModal">Add Recipe</a>
              <h4 className="container__myrecipes">My Recipes</h4><br /><br />
              <RecipeModal addRecipe={this.props.addRecipe}
                recipeCategories={this.props.categories}/>
            </div>

            <div className="row">
              {
                this.props.userRecipes.length > 0 ?
                  this.props.userRecipes.map((elem, index) => (
                    <RecipeCard key={elem.id} user={this.props.user}
                      recipe={elem} id={elem.id} onDelete={this.deleteRecipe}
                      editRecipe={this.editRecipe}
                    />)) :
                  <Exclamation>
                    <p className="text-muted">Sorry you haven't added any
                     recipe yet, please add to get started</p>
                  </Exclamation>
              }
            </div>
          </div>
        </main>
        <Pagination />
      </div>
    );
  }
}

/**
 * mapStateToProps
 * @param {any} state
 * @returns {object} object
 */
const mapStateToProps = (state) => ({
  recipes: state.recipes,
  user: state.auth.user,
  userRecipes: state.recipes.allRecipes,
  categories: state.recipes.recipeCategories
});

export default connect(mapStateToProps, {
  addRecipe, getUserRecipes, recipeCategories
})(UserRecipes);

