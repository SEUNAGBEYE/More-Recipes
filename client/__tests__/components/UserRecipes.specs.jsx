
import React from 'react';
import { shallow } from 'enzyme';
import { UserRecipes, mapStateToProps } from '../../main/src/components/recipes/UserRecipes';

const state = {
  auth: {
    user: {
      firstName: 'Seun',
      lastName: 'Agbeye'
    }
  },
  recipes: {
    recipeCategories: [{
      id: 1,
      name: 'Breakfast'
    }],
    popularRecipes: [{
      id: 1,
      name: 'Breakfast'
    }],
    allRecipes: [{
      id: 1,
      name: 'Breakfast'
    }],
  }
};

const recipe = {
  name: 'Amala',
  image: 'This is the image'
}


const props = {
  userRecipes: [recipe],
  categories: state.recipes.recipeCategories,
  recipesCount: 1,
  addRecipe: (() =>new Promise((resolve, reject) => {
    recipe ? resolve(recipe) : reject('Recipe not found');
  })),
  getUserRecipes: (() =>new Promise((resolve, reject) => {
    recipe ? resolve(recipe) : reject('Recipe not found');
  })),
  recipeCategories: jest.fn(),
  categories: [],
  history: {
    push: (() => {})
  },
  categories: [],
  user: {
    firstName: 'Seun',
    lastName: 'Agbeye'
  }
};

const event = {
  preventDefault: jest.fn(),
  target: {
    search: {
      value: '',
    },
    id: ''
  }
}

describe('# UserRecipes', () => {
  const wrapper = shallow(
      <UserRecipes {...props} />
  );
  it('should render successfully', (done) => {
    expect(wrapper).toBeDefined();
    expect(wrapper.getElement().type).toBe('div');
    expect(wrapper).toMatchSnapshot();
    done();
  });

  // it('should call mapStateToProps(state)', () => {
  //   mapStateToProps(state)
  // }); 
});
