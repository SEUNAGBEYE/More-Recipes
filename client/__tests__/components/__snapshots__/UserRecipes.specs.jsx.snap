// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# UserRecipes should render successfully 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserRecipes
    addRecipe={[Function]}
    categories={Array []}
    getUserRecipes={[Function]}
    history={
      Object {
        "push": [Function],
      }
    }
    recipeCategories={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    recipesCount={1}
    user={
      Object {
        "firstName": "Seun",
        "lastName": "Agbeye",
      }
    }
    userRecipes={
      Array [
        Object {
          "image": "This is the image",
          "name": "Amala",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <main
          style={
            Object {
              "marginTop": 40,
            }
          }
        >
          <div
            className="container"
          >
            <Connect(CategoryButton) />
          </div>
          <div
            className="container"
          >
            <div
              style={
                Object {
                  "marginTop": 100,
                  "textAlign": "center",
                }
              }
            >
              <Link
                className="btn auth-button"
                data-target="#addModal"
                data-toggle="modal"
                id="addRecipe"
                replace={false}
                to="#"
              >
                Add Recipe
              </Link>
              <h4
                className="container__myrecipes"
              >
                My Recipes
              </h4>
              <br />
              <br />
              <RecipeModal
                addRecipe={[Function]}
                recipeCategories={Array []}
              />
            </div>
            <div
              className="row"
            >
              <Connect(RecipeCard)
                editRecipe={undefined}
                id={undefined}
                onDelete={undefined}
                recipe={
                  Object {
                    "image": "This is the image",
                    "name": "Amala",
                  }
                }
                user={
                  Object {
                    "firstName": "Seun",
                    "lastName": "Agbeye",
                  }
                }
              />
            </div>
          </div>
        </main>,
        "",
      ],
      "id": "userRecipesBody",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="container"
            >
              <Connect(CategoryButton) />
            </div>,
            <div
              className="container"
            >
              <div
                style={
                  Object {
                    "marginTop": 100,
                    "textAlign": "center",
                  }
                }
              >
                <Link
                  className="btn auth-button"
                  data-target="#addModal"
                  data-toggle="modal"
                  id="addRecipe"
                  replace={false}
                  to="#"
                >
                  Add Recipe
                </Link>
                <h4
                  className="container__myrecipes"
                >
                  My Recipes
                </h4>
                <br />
                <br />
                <RecipeModal
                  addRecipe={[Function]}
                  recipeCategories={Array []}
                />
              </div>
              <div
                className="row"
              >
                <Connect(RecipeCard)
                  editRecipe={undefined}
                  id={undefined}
                  onDelete={undefined}
                  recipe={
                    Object {
                      "image": "This is the image",
                      "name": "Amala",
                    }
                  }
                  user={
                    Object {
                      "firstName": "Seun",
                      "lastName": "Agbeye",
                    }
                  }
                />
              </div>
            </div>,
          ],
          "style": Object {
            "marginTop": 40,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(CategoryButton) />,
              "className": "container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  style={
                    Object {
                      "marginTop": 100,
                      "textAlign": "center",
                    }
                  }
                >
                  <Link
                    className="btn auth-button"
                    data-target="#addModal"
                    data-toggle="modal"
                    id="addRecipe"
                    replace={false}
                    to="#"
                  >
                    Add Recipe
                  </Link>
                  <h4
                    className="container__myrecipes"
                  >
                    My Recipes
                  </h4>
                  <br />
                  <br />
                  <RecipeModal
                    addRecipe={[Function]}
                    recipeCategories={Array []}
                  />
                </div>,
                <div
                  className="row"
                >
                  <Connect(RecipeCard)
                    editRecipe={undefined}
                    id={undefined}
                    onDelete={undefined}
                    recipe={
                      Object {
                        "image": "This is the image",
                        "name": "Amala",
                      }
                    }
                    user={
                      Object {
                        "firstName": "Seun",
                        "lastName": "Agbeye",
                      }
                    }
                  />
                </div>,
              ],
              "className": "container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Link
                      className="btn auth-button"
                      data-target="#addModal"
                      data-toggle="modal"
                      id="addRecipe"
                      replace={false}
                      to="#"
                    >
                      Add Recipe
                    </Link>,
                    <h4
                      className="container__myrecipes"
                    >
                      My Recipes
                    </h4>,
                    <br />,
                    <br />,
                    <RecipeModal
                      addRecipe={[Function]}
                      recipeCategories={Array []}
                    />,
                  ],
                  "style": Object {
                    "marginTop": 100,
                    "textAlign": "center",
                  },
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Add Recipe",
                      "className": "btn auth-button",
                      "data-target": "#addModal",
                      "data-toggle": "modal",
                      "id": "addRecipe",
                      "replace": false,
                      "to": "#",
                    },
                    "ref": null,
                    "rendered": "Add Recipe",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "My Recipes",
                      "className": "container__myrecipes",
                    },
                    "ref": null,
                    "rendered": "My Recipes",
                    "type": "h4",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "addRecipe": [Function],
                      "recipeCategories": Array [],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Connect(RecipeCard)
                      editRecipe={undefined}
                      id={undefined}
                      onDelete={undefined}
                      recipe={
                        Object {
                          "image": "This is the image",
                          "name": "Amala",
                        }
                      }
                      user={
                        Object {
                          "firstName": "Seun",
                          "lastName": "Agbeye",
                        }
                      }
                    />,
                  ],
                  "className": "row",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "editRecipe": undefined,
                      "id": undefined,
                      "onDelete": undefined,
                      "recipe": Object {
                        "image": "This is the image",
                        "name": "Amala",
                      },
                      "user": Object {
                        "firstName": "Seun",
                        "lastName": "Agbeye",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "main",
      },
      "",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <main
            style={
              Object {
                "marginTop": 40,
              }
            }
          >
            <div
              className="container"
            >
              <Connect(CategoryButton) />
            </div>
            <div
              className="container"
            >
              <div
                style={
                  Object {
                    "marginTop": 100,
                    "textAlign": "center",
                  }
                }
              >
                <Link
                  className="btn auth-button"
                  data-target="#addModal"
                  data-toggle="modal"
                  id="addRecipe"
                  replace={false}
                  to="#"
                >
                  Add Recipe
                </Link>
                <h4
                  className="container__myrecipes"
                >
                  My Recipes
                </h4>
                <br />
                <br />
                <RecipeModal
                  addRecipe={[Function]}
                  recipeCategories={Array []}
                />
              </div>
              <div
                className="row"
              >
                <Connect(RecipeCard)
                  editRecipe={undefined}
                  id={undefined}
                  onDelete={undefined}
                  recipe={
                    Object {
                      "image": "This is the image",
                      "name": "Amala",
                    }
                  }
                  user={
                    Object {
                      "firstName": "Seun",
                      "lastName": "Agbeye",
                    }
                  }
                />
              </div>
            </div>
          </main>,
          "",
        ],
        "id": "userRecipesBody",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="container"
              >
                <Connect(CategoryButton) />
              </div>,
              <div
                className="container"
              >
                <div
                  style={
                    Object {
                      "marginTop": 100,
                      "textAlign": "center",
                    }
                  }
                >
                  <Link
                    className="btn auth-button"
                    data-target="#addModal"
                    data-toggle="modal"
                    id="addRecipe"
                    replace={false}
                    to="#"
                  >
                    Add Recipe
                  </Link>
                  <h4
                    className="container__myrecipes"
                  >
                    My Recipes
                  </h4>
                  <br />
                  <br />
                  <RecipeModal
                    addRecipe={[Function]}
                    recipeCategories={Array []}
                  />
                </div>
                <div
                  className="row"
                >
                  <Connect(RecipeCard)
                    editRecipe={undefined}
                    id={undefined}
                    onDelete={undefined}
                    recipe={
                      Object {
                        "image": "This is the image",
                        "name": "Amala",
                      }
                    }
                    user={
                      Object {
                        "firstName": "Seun",
                        "lastName": "Agbeye",
                      }
                    }
                  />
                </div>
              </div>,
            ],
            "style": Object {
              "marginTop": 40,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(CategoryButton) />,
                "className": "container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    style={
                      Object {
                        "marginTop": 100,
                        "textAlign": "center",
                      }
                    }
                  >
                    <Link
                      className="btn auth-button"
                      data-target="#addModal"
                      data-toggle="modal"
                      id="addRecipe"
                      replace={false}
                      to="#"
                    >
                      Add Recipe
                    </Link>
                    <h4
                      className="container__myrecipes"
                    >
                      My Recipes
                    </h4>
                    <br />
                    <br />
                    <RecipeModal
                      addRecipe={[Function]}
                      recipeCategories={Array []}
                    />
                  </div>,
                  <div
                    className="row"
                  >
                    <Connect(RecipeCard)
                      editRecipe={undefined}
                      id={undefined}
                      onDelete={undefined}
                      recipe={
                        Object {
                          "image": "This is the image",
                          "name": "Amala",
                        }
                      }
                      user={
                        Object {
                          "firstName": "Seun",
                          "lastName": "Agbeye",
                        }
                      }
                    />
                  </div>,
                ],
                "className": "container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <Link
                        className="btn auth-button"
                        data-target="#addModal"
                        data-toggle="modal"
                        id="addRecipe"
                        replace={false}
                        to="#"
                      >
                        Add Recipe
                      </Link>,
                      <h4
                        className="container__myrecipes"
                      >
                        My Recipes
                      </h4>,
                      <br />,
                      <br />,
                      <RecipeModal
                        addRecipe={[Function]}
                        recipeCategories={Array []}
                      />,
                    ],
                    "style": Object {
                      "marginTop": 100,
                      "textAlign": "center",
                    },
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Add Recipe",
                        "className": "btn auth-button",
                        "data-target": "#addModal",
                        "data-toggle": "modal",
                        "id": "addRecipe",
                        "replace": false,
                        "to": "#",
                      },
                      "ref": null,
                      "rendered": "Add Recipe",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "My Recipes",
                        "className": "container__myrecipes",
                      },
                      "ref": null,
                      "rendered": "My Recipes",
                      "type": "h4",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "br",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "br",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "addRecipe": [Function],
                        "recipeCategories": Array [],
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <Connect(RecipeCard)
                        editRecipe={undefined}
                        id={undefined}
                        onDelete={undefined}
                        recipe={
                          Object {
                            "image": "This is the image",
                            "name": "Amala",
                          }
                        }
                        user={
                          Object {
                            "firstName": "Seun",
                            "lastName": "Agbeye",
                          }
                        }
                      />,
                    ],
                    "className": "row",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "editRecipe": undefined,
                        "id": undefined,
                        "onDelete": undefined,
                        "recipe": Object {
                          "image": "This is the image",
                          "name": "Amala",
                        },
                        "user": Object {
                          "firstName": "Seun",
                          "lastName": "Agbeye",
                        },
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "main",
        },
        "",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
